provd/internal/services/user/user.go

```diff
@@ -186,51 +201,83 @@ func (s *Service) ValidateUsername(ctx context.Context, req *pb.ValidateUsername
 		return &pb.ValidateUsernameResponse{UsernameValidation: pb.UsernameValidation_TOO_LONG}, nil
 	}
 
-	// Check if username is in reserved list
-	// Read line by line to avoid loading the whole file into memory
-	file, err := reservedUsernamesFS.Open("reserved-usernames")
+	// Open the passwd.master file
+	passwdFile, err := os.Open("/usr/share/base-passwd/passwd.master")
```

Comment:
How do you mock that in tests?

Trigger cases when there is no file, files with incorrect parsing and so on?

Reply:
We could have the filepaths configurable as options on the User service struct?

Reply:
exactly!